### ISSUE TITLE
NaN loss after a few epochs

### ISSUE THREAD

**Issue by groadabike** [borchero/pycave#3]
Hi @borchero,
I am training a GMM using a 40-dimensional feature vector.
There is no problem when I train the model using the features as is. However, if I standardise the features using sklearn's StandardScaler, I got nan loss after three of four epochs.

Is there any way I can easily debug the model/engine to try to find the source of the issue?

**Comment by borchero** [comment]
If you want it to debug within PyCave, I would recommend that you add some print statements to `pycave/bayes/gmm/engine.py` in the `train_batch` method. Otherwise, I can also have a look if you can tell me how I can replicate your data (I suppose every feature is distributed according to the Standard Normal then?).

**Comment by groadabike** [comment]
Hi @borchero ,
It seems that my problem was the singularity GMM issue.
I solved the problem by increasing the covariance regularisation factor from 1e-6 to 1e-5.

I found that when I apply StandardScaler to my data (audio MFCCs) and select more than 20 components, one component ended with a covariance very close to the regularization factor, i.e., all its values close to 1e-6.
Using the same logic as in Sklearn, I increase that factor (https://github.com/scikit-learn/scikit-learn/blob/b3ea3ed6a09fe774dfc5160a65172b1bacbb2a82/sklearn/mixture/_gaussian_mixture.py#L306).

Is there a reason why the regularisation factor is not a model parameter?
